### ISSUE TITLE
YSlow does not load in Firebug 1.10.1 and Firefox 15

### ISSUE THREAD

**Issue by codeuniquely** [marcelduran/yslow#50]
Clicking the YSlow icon opens a smal panel at the bottom of the browser with a Run YSlow button that does nothing and a close button that works. YSlow menu header no longer shows in Firebug menus. Have un-installed and re-installed but this does not fix the issue.

**Comment by sigviper** [comment]
Same here. Firebug 1.10.1 and Firefox 14.0.1 (both Windows XP and Ubuntu).
The windows one has only two extensions installed: Firebug and Yslow.

**Comment by marcelduran** [comment]
If anyone can take a look at https://github.com/marcelduran/yslow/blob/da3a53c21dbb9651e967a74a9e96a720796a0050/src/firefox/chrome/content/yslow-firefox.js#L80 this is where YSlow is hooked into Firebug, I really appreciate it. Thanks.

**Comment by jphpsf** [comment]
I am seeing the issue here as well. Firebug 1.10.2 and Firefox 14.0.1.

1) With Firebug being closed, if I click on the YSlow icon in the status bar,  I see the following error in the error log:

```
Timestamp: 08/03/2012 01:58:24 PM
Error: TypeError: win is undefined
Source File: chrome://yslow/content/yslow/controller.js
Line: 78
```

2) When Firebug is open, I do not see the YSlow panel. It looks like Firebug 1.10 has changed the API to add buttons to the panel toolbar. I am wondering if that is the cause of the issue?

See [this example](https://code.google.com/p/fbug/source/browse/examples/firebug1.10/Toolbar/chrome/content/myPanel.js).

It looks like a Firebug extension must define the `getPanelToolbarButtons` method. I grepped the YSlow code and did not see any reference.

I don't have an extension dev environment setup yet, so I can not really investigate at the moment. But I'll give it a shot :) Don't hold your breath too long though, I need to learn how Firebug works and the YSlow code base ;)

**Comment by marcelduran** [comment]
Thanks @jphpsf, you might also want to look at https://github.com/marcelduran/yslow/blob/master/src/firefox/chrome/content/yslow-firebug.js which is where Firebug initializer methods are called, they were based in Honza's old Extending Firebug tutorial: http://www.softwareishard.com/blog/firebug-tutorial/extending-firebug-hello-world-part-i/

**Comment by marcelduran** [comment]
New attempt to fix, awaiting reviewers to release [3.1.4](https://addons.mozilla.org/en-us/firefox/addon/yslow/versions/?page=1#version-3.1.4). See [commit](https://github.com/marcelduran/yslow/commit/d6508e6a16040a763da9caa6676241443fb21540) changes. Thanks for your patience.
